include/revng/Yield/Support/GraphLayout/SugiyamaStyle/InternalGraph.h

```diff
@@ -0,0 +1,234 @@
+#pragma once
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include <unordered_map>
+#include <variant>
+
+#include "revng/ADT/GenericGraph.h"
+#include "revng/Yield/Support/GraphLayout/Traits.h"
+
+namespace yield::layout::sugiyama {
+
+class InternalGraph;
+
+namespace detail {
+
+struct InternalNode {
+  layout::Point Center;
+  layout::Size Size;
+
+  bool IsVirtual;
+
+private:
+  std::size_t Index;
+
+private:
+  friend InternalGraph;
```

Comment:
Why do you need this?

Reply:
Because they require an index to be created. And since I was unable to keep them `const` (we require an assignment operator), I softened the restriction to "only `InternalGraph` (the one who keeps track of the used indices) can make these).